src/core/Object.cpp

```diff
@@ -133,17 +80,20 @@ void Object::write_objects_map_to( std::ostream& out, object_map_t* map ) {
 #endif
 		return;
 	}
-
+	object_map_t snapshot;
 	if ( map == nullptr ) {
-		map = &__objects_map;
+		snapshot = getObjectMap();
+		map = &snapshot;
 	}
 	
 	std::ostringstream o;
 	pthread_mutex_lock( &__mutex );
 	object_map_t::iterator it = map->begin();
 	while ( it != map->end() ) {
-		o << "\t[ " << std::setw( 30 ) << ( *it ).first << " ]\t" << std::setw( 6 ) << ( *it ).second.constructed << "\t" << std::setw( 6 ) << ( *it ).second.destructed
-		  << "\t" << std::setw( 6 ) << ( *it ).second.constructed - ( *it ).second.destructed << std::endl;
+		if ( it->second.constructed || it->second.destructed ) {
```

Comment:
Shouldn't this be a `&&`?

Reply:
> Hey @Floessie,
> 
> Thanks for the insights!
> 
> > Are you aware that double underscores prefixes are reserved for the compiler/library implementation?
> 
> Well, TBH I wasn't aware of the exact reason its usage is a no-no but there is already a rough coding standard in the [DEVELOPERS](https://github.com/hydrogen-music/hydrogen/blob/master/DEVELOPERS) file telling to use `m_*` to indicate member variables in Hydrogen instead. Thing is, some parts of the code are really old (we don't know for sure how old but the first working version was released more than 20 years ago). So, it gets more rocky the further you descend. Our current policy is to only refactor it when rewriting larger parts (during development). It's a little bit more on the "don't change a running system" than on the modernization side of the tradeoff since our goal is stability over new features.

About the project's coding standards, I think they could use a little refresher. I find this utterly ugly for instance. 
```c++
if( condition ){
  ...
}
```
I mean, control flow are not functions.
```c++
if ( condition ) {
  ...
}
```
please :grin:

Reply:
> Shouldn't this be a `&&`?

In fact, if something goes plain wrong, `||` is better in this case. Suppose we allocate some object of some type and never delete it.